pkg/aggregator/ckey/key_test.go

```diff
@@ -49,6 +49,57 @@ func TestCompare(t *testing.T) {
 	assert.False(t, Equals(base, diff))
 }
 
+func TestTagsOrderAndDupsDontMatter(t *testing.T) {
+	assert := assert.New(t)
+
+	name := "metrics.to.test.hashing"
+	hostname := "hostname.localhost"
+	tags := []string{"bar", "foo", "key:value", "key:value2"}
+
+	generator := NewKeyGenerator()
+	key, _ := generator.Generate(name, hostname, tags)
+
+	// change tags order, the generated key should be the same
+	tags[0], tags[1], tags[2], tags[3] = tags[3], tags[0], tags[1], tags[2]
+	key2, _ := generator.Generate(name, hostname, tags)
+	assert.Equal(key, key2, "order of tags should not matter")
+
+	// add a duplicated tag
+	tags = append(tags, "key:value", "foo")
+	key3, _ := generator.Generate(name, hostname, tags)
+	assert.Equal(key, key3, "duplicated tags should not matter")
+
+	// and now, completely change of the tag, the generated key should NOT be the same
+	tags[2] = "another:tag"
+	key4, _ := generator.Generate(name, hostname, tags)
+	assert.NotEqual(key, key4, "tags content should matter")
+}
```

Comment:
I added another test to give a bit more iterations to this testing.  It doesn't necessarily hit all of the corner cases (hash collisions, zero hashes, etc.), but it did find the issue with >512 tags.

```suggestion
}

func TestTagsAreDedupedWhileGeneratingCKey(t *testing.T) {
        withSizeAndSeed := func(size, iterations int, seed int64) func(*testing.T) {
                return func(t *testing.T) {
                        assert := assert.New(t)
                        r := rand.New(rand.NewSource(seed))

                        name := "metrics.to.test.hashing"
                        hostname := "hostname.localhost"
                        tags := genTags(size)

                        generator := NewKeyGenerator()
                        expKey, _ := generator.Generate(name, hostname, tags)

                        for i := 0; i < iterations; i++ {
                                r.Shuffle(size, func(i, j int) { tags[i], tags[j] = tags[j], tags[i] })
                                key, _ := generator.Generate(name, hostname, tags)
                                assert.Equal(expKey, key, "order of tags should not matter")
                        }
                }
        }
        t.Run("smallish", withSizeAndSeed(10, 200, 0x398192f0a9c0))
        t.Run("bigger", withSizeAndSeed(50, 100, 0x398192f0a9c0))
        t.Run("huge", withSizeAndSeed(600, 10, 0x398192f0a9c0))
}
```